### PR TITLE
Add basic filter panel to Search and Filter component

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done 

Add basic filter panel to Search and Filter component

## QA

- Fire up Storybook: http://localhost:65491/?path=/story/searchandfilter--default-story
- Click in input to activate panel
- Press Escape to close
- Click outside input or panel to close

Fixes https://github.com/canonical-web-and-design/juju-squad/issues/1335

### CODE REVIEW

File: src/components/SearchAndFilter/SearchAndFilter.js [hunk 1]
```diff
@@ -13,14 +18,56 @@ const SearchAndFilter = ({ externallyControlled = false, onChange }) => {
...
         externallyControlled={externallyControlled}
         placeholder="Search and filter"
         onChange={(searchTerm) => searchOnChange(searchTerm)}
+        onFocus={() => setFilterPanelVisible(true)}
         value={searchTerm}
       />
+      <div
+        className="search-and-filter__panel"
+        ref={filterPanelRef}
+        data-visible={filterPanelVisible}
```
Comment (anthonydillon) [inline src/components/SearchAndFilter/SearchAndFilter.js:67]: Why not use `aria-hidden` and make it accessible from the get go?

File: src/components/SearchAndFilter/SearchAndFilter.scss [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  .p-contextual-menu__dropdown {
+    top: 0;
+    left: 0;
+  }
+
+  &__panel {
+    padding: 0;
+    transition: opacity 0.25s;
+
+    &[data-visible="false"] {
```
Comment (anthonydillon) [inline src/components/SearchAndFilter/SearchAndFilter.scss:28]: Same here you should target `aria-hidden`

File: src/components/SearchAndFilter/SearchAndFilter.test.js [hunk 3]
```diff
@@ -22,4 +22,18 @@ describe("Search and filter ", () => {
     wrapper.find(".p-search-box__input").simulate("change");
     expect(mockOnChange).toHaveBeenCalledTimes(1);
   });
+
+  it("shows panel on focus", () => {
```
Comment (anthonydillon) [inline src/components/SearchAndFilter/SearchAndFilter.test.js:26]: Can you add a test for the click or is that difficult?

File: src/components/SearchAndFilter/SearchAndFilter.scss [hunk 4]
```diff
@@ -0,0 +1,33 @@
+.search-and-filter {
+  position: relative;
```
Comment (anthonydillon) [inline src/components/SearchAndFilter/SearchAndFilter.scss:2]: I don't see any internal position so I don't think we need this?

**Comment by anthonydillon** [comment]
## QA 
- If you click on the field the panel is revealed but clicking on the field again closes it and it doesn't open again.
- If you click the field the panel is revealed but if you tab off you can't hit Esc to close the panel.

File: src/components/SearchAndFilter/SearchAndFilter.js (hunk 1 above)
Comment (barrymcgee) [reply src/components/SearchAndFilter/SearchAndFilter.js:67]: Good call 🙌🏻

File: src/components/SearchAndFilter/SearchAndFilter.test.js (hunk 3 above)
Comment (barrymcgee) [reply src/components/SearchAndFilter/SearchAndFilter.test.js:26]: I don't think we need to test that the browser adds focus on click?

Comment (barrymcgee) [reply src/components/SearchAndFilter/SearchAndFilter.test.js:26]: Gotcha.. added

**Pull request merged**